var/spack/repos/builtin/packages/py-h5py/package.py

```diff
@@ -61,7 +63,8 @@ class PyH5py(PythonPackage):
     # Link dependencies (py-h5py v2 cannot build against HDF5 1.12 regardless
```

Comment:
Numpy requirements also need to be loosened, not just in 3.8.0 but in older versions too

Reply:
Looks like they applied until 3.5. I've included the necessary changes.